pymc/tests/test_distributions_timeseries.py

```diff
@@ -88,16 +88,15 @@ def test_grw_inference(self):
         with pm.Model():
             _mu = pm.Uniform("mu", -10, 10)
             _sigma = pm.Uniform("sigma", 0, 10)
-            grw = GaussianRandomWalk("grw", _mu, _sigma, steps=steps, observed=obs)
-
-            with pytest.raises(TypeError) as err:
-                trace = pm.sample()
+            # Workaround for bug in `at.diff` when data is constant
+            obs_data = pm.MutableData("obs_data", obs)
```

Comment:
@ricardoV94 Just curious how this gets around the bug. What does the MutableData object change in the Aesara logic?

Reply:
`MutableData` uses a `SharedVariable` whose size is allowed to change. The Aesara bug happens when you have a constant whose size is not allowed to change anymore and it wrongly guesses what the new output size will be (it forgets that it is (size-1). I'll see if I can patch it today in Aesara

Reply:
If you change can you cc on the pr, or link here. it'll add to the learning